contracts/common-utils/src/math.rs

```
@@ -225,3 +225,136 @@ pub fn compound_interest(
+/// # Arguments
+/// * `values` - Slice of fixed-point values
+/// * `mean` - The mean of the values as a fixed-point number
+///
+/// # Returns
+/// The standard deviation as a fixed-point number
+pub fn standard_deviation(values: &[i128], mean: i128) -> core::result::Result<i128, CommonError> {
+    let var = variance(values, mean)?;
+    
+    // Square root approximation using Newton's method
+    let mut x = var;
+    let mut prev = 0i128;
```

Clippy (1.97.1)

[warning] value assigned to `prev` is never read

(warning)